tests/fixtures/application_fixtures.py

```diff
@@ -839,7 +848,12 @@ def succeeded_privacy_request(cache, db: Session, policy: Policy) -> PrivacyRequ
             "client_id": policy.client_id,
         },
     )
-    pr.cache_identity({"email": "email@example.com"})
+    identity_kwargs = {"email": "email@example.com"}
+    pr.cache_identity(identity_kwargs)
+    pr.persist_identity(
```

Comment:
Bug: Policy webhooks can have derived_identities returned. Neither `PrivacyRequest.trigger_policy_webhook` nor `privacy_request_endpoints > resume_privacy_request` which both update the identity graph, persist the data to the database, they only add it to the redis cache.

Reply:
> Would it be useful to have a method that both persists the identity in the cache and in the database at the same time?

I'm torn here. On the one hand it's nice to have consistency, on the other, as you rightly suggest above, it means we'll need to be plumbing the DB connection in more places. I'm not sure if it's better to have the execution update the cache with identity data at the very start before the traversal, such that we can guarantee the traversal will always use what was provided by the user on privacy request creation. That way the internals can still use the cache and benefit from the speed, and less refactoring is required.

Reply:
> Bug: Policy webhooks can have derived_identities returned. Neither PrivacyRequest.trigger_policy_webhook nor privacy_request_endpoints > resume_privacy_request which both update the identity graph, persist the data to the database, they only add it to the redis cache.

We should these concerns for now. The `ProvidedIdentity` is useful for facilitating request search based on the exact identity provided, we don't currently want to search based on derived identities, or return them into the UI, so should be careful when we update the `ProvidedIdentity` table as that's what will get displayed in the UI (and doesn't currently support anything beyond `email` and `phone_number`).

Reply:
>  That way the internals can still use the cache and benefit from the speed, and less refactoring is required. What do you think?

Thinking about this more, I agree, it's in line with our original design, in that we query everything up front, build the graph, and execute it. We're not regularly querying the database as we execute the traversal which I think is good for performance.

Reply:
> We should separate these concerns for now. The ProvidedIdentity is useful for facilitating request search based on the exact identity provided, we don't currently want to search based on derived identities

OK, that makes sense